p2p-sync/src/main/java/org/smartregister/p2p/activity/P2pModeSelectActivity.java

```diff
@@ -0,0 +1,315 @@
+package org.smartregister.p2p.activity;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.IntentSender;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.common.api.ResolvableApiException;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.LocationSettingsResponse;
+import com.google.android.gms.location.LocationSettingsStatusCodes;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+import org.smartregister.p2p.R;
+import org.smartregister.p2p.contract.P2pModeSelectContract;
+import org.smartregister.p2p.dialog.StartDiscoveringModeProgressDialog;
+import org.smartregister.p2p.dialog.StartReceiveModeProgressDialog;
+import org.smartregister.p2p.handler.OnActivityRequestPermissionHandler;
+import org.smartregister.p2p.handler.OnActivityResultHandler;
+import org.smartregister.p2p.handler.OnResumeHandler;
+import org.smartregister.p2p.interactor.P2pModeSelectInteractor;
+import org.smartregister.p2p.presenter.P2pModeSelectPresenter;
+import org.smartregister.p2p.util.Constants;
+import org.smartregister.p2p.util.DialogUtils;
+import org.smartregister.p2p.util.Permissions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import timber.log.Timber;
+
+public class P2pModeSelectActivity extends AppCompatActivity implements P2pModeSelectContract.View {
+
+    private Button sendButton;
+    private Button receiveButton;
+
+    private P2pModeSelectContract.Presenter presenter;
+    private ArrayList<OnActivityResultHandler> onActivityResultHandlers = new ArrayList<>();
+    private ArrayList<OnResumeHandler> onResumeHandlers = new ArrayList<>();
+    private ArrayList<OnActivityRequestPermissionHandler> onActivityRequestPermissionHandlers = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_p2p_mode_select);
+
+        sendButton = findViewById(R.id.btn_p2pModeSelectActivity_send);
+        receiveButton = findViewById(R.id.btn_p2pModeSelectActivity_receive);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        initializePresenter();
+
+        sendButton.setOnClickListener(null);
+        receiveButton.setOnClickListener(null);
```

Comment:
Is this needed
```
        sendButton.setOnClickListener(null);
        receiveButton.setOnClickListener(null);
```

Reply:
The activity is/was behaving kinda weird whereby the `onCreate` method is called twice everytime it starts. This also weirdly resulted in the `presenter.onSendButtonClicked()` being called twice (😞 which shouldn't be remotely possible) I think that this should remedy the situation. I'll retest it and get back to you

Reply:
After testing on my device `onCreate` and `onStart` is only invoked only once.

Shouldn't the click handlers be setup in `onCreate`?
Also if you set them up on `onStart` then you should unregister then on `onStop`

Reply:
Cool